Source/CsDebugScript.Engine/Debugger.cs

```diff
@@ -629,6 +629,67 @@ public static void Terminate(Process process)
         }
+
+        /// <summary>
+        /// Sets breakpoint in specified process by evaluating given expression.
+        /// Breakpoint breaks exeuction until debugger is released.
+        /// </summary>
+        /// <param name="process">Process in which breakpoint is to be set.</param>
+        /// <param name="expression">Expression that evaulates into breakpoint location.</param>
+        /// <returns>Newly created breakpoint.</returns>
+        public static IBreakpoint SetBreakpoint(Process process, string expression)
+        {
+            return SetBreakpoint(process, expression, () => BreakpointEventStatus.BreakDebugger);
+        }
+
+        /// <summary>
+        /// Sets breakpoint in current process by evaulating given expression.
+        /// Breakpoint breaks exeuction until debugger is released.
+        /// </summary>
+        /// <param name="expression">Expression that evaulates into breakpoint location.</param>
+        /// <returns>Newly created breakpoint.</returns>
+        public static IBreakpoint SetBreakpoint(string expression)
+        {
+            return SetBreakpoint(expression, () => BreakpointEventStatus.BreakDebugger);
+        }
+
+        /// <summary>
+        /// Sets breakpoint in specified process by evaluating given expression.
+        /// On breakpoint hit given action is executed.
+        /// Action return code specifies the state of debugger after breakpoint
+        /// is processed.
+        /// </summary>
+        /// <param name="expression">Expression that evaulates into breakpoint location.</param>
+        /// <param name="action">Action to be executed when breakpoint is hit.</param>
+        /// <returns>Newly created breakpoint.</returns>
+        public static IBreakpoint SetBreakpoint(string expression, Func<BreakpointEventStatus> action)
+        {
+            return Context.Debugger.SetBreakpoint(Process.Current, expression, action);
+        }
+
+        /// <summary>
+        /// Sets breakpoint in specified process by evaluating given expression.
+        /// On breakpoint hit given action is executed.
+        /// Action return code specifies the state of debugger after breakpoint
+        /// is processed.
+        /// </summary>
+        /// <param name="process">Process in which breakpoint is to be set.</param>
+        /// <param name="expression">Expression that evaulates into breakpoint location.</param>
+        /// <param name="action">Action to be executed when breakpoint is hit.</param>
+        /// <returns>Newly created breakpoint.</returns>
+        public static IBreakpoint SetBreakpoint(Process process, string expression, Func<BreakpointEventStatus> action)
+        {
+            return Context.Debugger.SetBreakpoint(process, expression, action);
+        }
+
+        /// <summary>
+        /// Returns all breakpoints for this session.
+        /// </summary>
+        /// <returns></returns>
+        public static IEnumerable<IBreakpoint> GetAllBreakpoints()
```

Comment:
>GetAllBreakpoints [](start = 47, length = 17)

Interface for setting break point is per process. How can I filter breakpoint only for specified process?